cpp/src/strings/convert.cu

```diff
@@ -0,0 +1,257 @@
+{
+namespace
+{
+
+/**
+ * @brief Converts a single string into an integer.
+ * The '+' and '-' are allowed but only at the beginning of the string.
+ * The string is expected to contain base-10 [0-9] characters only.
+ * Any other character will end the parse.
+ * Overflow of int64 type is not detected.
+ */
+__device__ int64_t string_to_integer( const string_view& d_str )
+{
+    int64_t value = 0;
+    size_type bytes = d_str.size_bytes();
+    const char* ptr = d_str.data();
+    int sign = 1;
+    if( *ptr == '-' || *ptr == '+' )
+    {
+        sign = (*ptr=='-' ? -1:1);
+        ++ptr;
+        --bytes;
+    }
+    for( size_type idx=0; idx < bytes; ++idx )
+    {
+        char chr = *ptr++;
+        if( chr < '0' || chr > '9' )
+            break;
+        value = (value * 10) + static_cast<int64_t>(chr - '0');
+    }
+    return value * static_cast<int64_t>(sign);
+}
+
+} // namespace
+
+//
+std::unique_ptr<cudf::column> to_integers( strings_column_view strings,
+                                           rmm::mr::device_memory_resource* mr,
+                                           cudaStream_t stream)
+{
+    size_type strings_count = strings.size();
+    if( strings_count == 0 )
+        return std::make_unique<column>( data_type{INT32}, 0,
+                          rmm::device_buffer{0,stream,mr}, // data
+                          rmm::device_buffer{0,stream,mr}, 0 ); // nulls
+
+    auto execpol = rmm::exec_policy(stream);
+    auto strings_column = column_device_view::create(strings.parent(), stream);
+    auto d_column = *strings_column;
+
+    // copy null mask
+    rmm::device_buffer null_mask;
+    cudf::size_type null_count = d_column.null_count();
+    if( d_column.nullable() )
+        null_mask = rmm::device_buffer( d_column.null_mask(),
+                                        gdf_valid_allocation_size(strings_count),
+                                        stream, mr);
+    // create output column
+    auto results = std::make_unique<cudf::column>( cudf::data_type{cudf::INT32}, strings_count,
+        rmm::device_buffer(strings_count * sizeof(int32_t), stream, mr),
+        null_mask, null_count);
+    auto results_view = results->mutable_view();
+    auto d_results = results_view.data<int32_t>();
+    // set the values
+    thrust::transform( execpol->on(stream),
+        thrust::make_counting_iterator<size_type>(0),
+        thrust::make_counting_iterator<size_type>(strings_count),
+        d_results,
+        [d_column] __device__ (size_type idx) {
+            if( d_column.is_null(idx) )
+                return int32_t(0);
+            return static_cast<int32_t>(string_to_integer(d_column.element<cudf::string_view>(idx)));
+        });
+    results->set_null_count(null_count);
+    return results;
+}
+
+namespace
+{
+
+/**
+ * @brief Calculate the size of the each string required for
+ * converting each integer in base-10 format.
+ */
+template <typename IntegerType>
+struct integer_to_string_size_fn
```

Comment:
What do you think about an implementation like this? https://stackoverflow.com/a/1498561/11341974

Integer division is expensive, so I'm thinking this would be faster.